Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -719,37 +688,33 @@ static bool sampleBufferRenderersSupportKeySession()
         weakThis->didProvideContentKeyRequestInitializationDataForTrackID(WTFMove(initData), trackID, nullptr);
     });
 
-    m_parsingSucceeded = true;
-
-    m_appendQueue->dispatch([data = WTFMove(data), weakThis = WeakPtr { *this }, parser = m_parser]() mutable {
-        parser->appendData(WTFMove(data), [weakThis = WTFMove(weakThis)]() mutable {
-            callOnMainThread([weakThis = WTFMove(weakThis)] {
-                if (weakThis)
-                    weakThis->appendCompleted();
-            });
-        });
+    return invokeAsync(m_appendQueue, [data = WTFMove(data), parser = m_parser]() mutable {
+        return parser->appendData(WTFMove(data));
```

Comment:
Ah I see why you changed to a promise.
If it is the only use of a promise, maybe you could promisify the synchronous result directly?

Reply:
No. All the callbacks are called asynchronously too